ui-trackoccupancydiagram/src/components/helpers/drawElements/drawOccupancyZones.ts

```diff
@@ -0,0 +1,61 @@
+import { drawOccupancyZonesTexts } from './drawOccupancyZonesTexts';
+import {
+  TRACK_HEIGHT_CONTAINER,
+  CANVAS_PADDING,
+  OCCUPANCY_ZONE_START,
+  OCCUPANCY_ZONE_HEIGHT,
+  COLORS,
+} from '../../consts';
+import type { OccupancyZone, Track } from '../../types';
+
+export const drawOccupancyZones = ({
+  ctx,
+  width,
+  height,
+  tracks,
+  occupancyZones,
+  getTimePixel,
+}: {
+  ctx: CanvasRenderingContext2D;
+  width: number;
+  height: number;
+  tracks: Track[] | undefined;
+  occupancyZones: OccupancyZone[] | undefined;
+  getTimePixel: (time: number) => number;
+}) => {
+  ctx.clearRect(0, 0, width, height);
+
+  tracks?.forEach((track, index) => {
+    const trackTranslate = index === 0 ? CANVAS_PADDING : TRACK_HEIGHT_CONTAINER;
+    ctx.translate(0, trackTranslate);
+
+    const filteredOccupancyZone = occupancyZones?.filter((zone) => zone.trackId === track.id);
```

Comment:
i think you can rename it by trackOccupancyZones and also you can do an early return if there is no zone

Reply:
done